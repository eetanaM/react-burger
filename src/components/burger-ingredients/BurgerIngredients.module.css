.burger_ingredients_container {
    position: relative;
    display: flex;
    flex-direction: column;

    max-width: 600px;
    height: calc(100vh - 128px);
}

.tab_menu {
    display: flex;
}

.ingredients_list {
    list-style-type: none;
    overflow-y: scroll;
    overflow-x: hidden;
    box-sizing: border-box;

    padding-inline-start: 0;
}

.ingredient_cards_container {
    display: flex;
    flex-flow: column wrap;
}

.ingredient_card > a {
    text-decoration: none;
    color: white
}

.ingredient_card {
    display: flex;
    flex-flow: row wrap;
    justify-content: start;
    align-items: start;

    width: 100%;
    row-gap: 32px;
    column-gap: 24px;
    text-align: center;
    composes: pt-6 ml-4 from global;
}
