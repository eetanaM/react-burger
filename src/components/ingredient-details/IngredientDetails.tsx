import React from "react";
import styles from "./IngredientDetails.module.css"
import { CloseIcon } from "@ya.praktikum/react-developer-burger-ui-components"

export default function IngredientDetails({ currentIngredient }:any) {
    return (
        <div className={`${styles.content_container} pr-10 pl-10`}>
            <div className={`${styles.content_header} pt-10`}>
                <h3 className="text text_type_main-large">Детали ингредиентов</h3>
                <CloseIcon type="primary"/>
            </div>
            <div className={`${styles.content_main} pl-10 pt-10 pr-10`}>
                <img src={currentIngredient.image} alt={currentIngredient.name} />
                <h3 className="text text_type_main-default">{currentIngredient.name}</h3>
                <ul className={styles.content_info_list}>
                    <li className="text text_type_main-default text_color_inactive">
                        <h4>Калории, ккал</h4>
                        <span>{currentIngredient.calories}</span>
                    </li>
                    <li className="text text_type_main-default text_color_inactive">
                        <h4>Белки, г</h4>
                        <span>{currentIngredient.proteins}</span>
                    </li>
                    <li className="text text_type_main-default text_color_inactive">
                        <h4>Жиры, г</h4>
                        <span>{currentIngredient.fat}</span>
                    </li>
                    <li className="text text_type_main-default text_color_inactive">
                        <h4>Углеводы, г</h4>
                        <span>{currentIngredient.carbohydrates}</span>
                    </li>
                </ul>
            </div>
        </div>
    )
}
