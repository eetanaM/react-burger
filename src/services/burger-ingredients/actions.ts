import { SerializedError } from "@reduxjs/toolkit"

const SERVER_URL: string = "https://norma.nomoreparties.space/api/ingredients"

export const getIngredients = async () => {
    try {
        const res = await fetch(SERVER_URL)
        if (!res.ok) {
            throw new Error("Request error occured!")
        }
        const ingredients = await res.json()
        return ingredients.data
    } catch (error: any) {
        return error
    }
}
