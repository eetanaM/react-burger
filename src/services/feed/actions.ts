import { createAction } from "@reduxjs/toolkit";


export const wsConnect = createAction<string, "ORDERS_CONNECT">("ORDERS_CONNECT");
export const wsDisconnect = createAction("ORDERS_DISCONNECT");
