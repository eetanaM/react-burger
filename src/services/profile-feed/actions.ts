import { createAction } from "@reduxjs/toolkit";


export const profileWsConnect = createAction<string, "PROFILE_ORDERS_CONNECT">("PROFILE_ORDERS_CONNECT");
export const profileWsDisconnect = createAction("PROFILE_ORDERS_DISCONNECT");
