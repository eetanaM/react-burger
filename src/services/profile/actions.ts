import { createAsyncThunk } from "@reduxjs/toolkit";

import {
    registerUser as registerAPI,
    loginUser as loginAPI,
    logoutUser as logoutAPI,
    getUserData as getUserDataAPI,
    configureUserData as configureUserDataAPI,
} from "../../utils/api"

import { ILogin, IRegister } from "../../utils/types/type";

export const registerUser = createAsyncThunk(
    "profile/registerUser",
    async (user: IRegister) => {
        return registerAPI(user.email, user.password, user.userName)
    }
)

export const loginUser = createAsyncThunk(
    "profile/loginUser",
    async (user: ILogin) => {
        return loginAPI(user.email, user.password)
    }
)

export const getUser = createAsyncThunk("profile/getUser", getUserDataAPI)

export const logoutUser = createAsyncThunk("profile/logoutUser", logoutAPI)

export const configureUser = createAsyncThunk(
    "profile/configureUser",
    async (user: IRegister) => {
        return configureUserDataAPI(user.email, user.password, user.userName)
    }
)
