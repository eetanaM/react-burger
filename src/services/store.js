import { configureStore as createStore } from "@reduxjs/toolkit";
import { rootReducer } from "./rootReducer";

export const store = createStore({
  reducer: rootReducer,
});
