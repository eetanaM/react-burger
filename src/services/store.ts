import { configureStore as createStore } from "@reduxjs/toolkit";
import { rootReducer } from "./rootReducer";

import customMiddleware from "./middleware/custom-middleware";
import { socketMiddleware } from "./middleware/socket-middleware";

import { wsConnect, wsDisconnect } from "./feed/actions";
import { wsError, wsMessage } from "./feed/slice";

import { profileWsConnect, profileWsDisconnect } from "./profile-feed/actions";
import { profileWsError, profileWsMessage } from "./profile-feed/slice";

import { IOrdersResponse } from "../utils/types/web-socket";

const feedMiddleware = socketMiddleware<unknown, IOrdersResponse>({
  connect: wsConnect,
  disconnect: wsDisconnect,
  onError: wsError,
  onMessage: wsMessage,
})

const profileFeedMiddleware = socketMiddleware<unknown, IOrdersResponse>({
  connect: profileWsConnect,
  disconnect: profileWsDisconnect,
  onError: profileWsError,
  onMessage: profileWsMessage,
})

export const store = createStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(customMiddleware, feedMiddleware, profileFeedMiddleware)
});


export type AppStore = typeof store
export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = typeof store.dispatch
