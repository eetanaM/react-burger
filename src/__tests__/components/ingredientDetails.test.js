import { create } from "react-test-renderer";
import { MemoryRouter, Route, Routes } from "react-router";
import { Provider } from "react-redux";
import { store } from "../../services/store";
import { loadIngredients } from "../../services/burger-ingredients/actions";
import IngredientDetails from "../../components/ingredient-details/IngredientDetails";

describe("check ingredientDetails component", () => {
  it("should be equal to snapshot", async () => {
    // Arrange (loading ingredients into store)
    await store.dispatch(loadIngredients());

    // Act (rendering certain ingredient's card)
    const renderedValue = create(
      <MemoryRouter initialEntries={["/ingredients/643d69a5c3f7b9001cfa093c"]}>
        <Provider store={store}>
          <Routes>
            <Route path="/ingredients">
              <Route path=":id" element={<IngredientDetails />} />
            </Route>
          </Routes>
        </Provider>
      </MemoryRouter>
    );

    // Assert (checking if the rendered component matches the snapshot)
    expect(renderedValue).toMatchSnapshot();
  });
});
