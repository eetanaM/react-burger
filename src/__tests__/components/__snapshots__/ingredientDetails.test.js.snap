// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check ingredientDetails component should be equal to snapshot 1`] = `
<div
  className="content_main"
>
  <h1
    className="content_main_header text text_type_main-large mt-10 pt-3 pb-3"
    data-testid="ingredient_details_header_test_element"
  >
    Детали ингредиента
  </h1>
  <img
    alt="Краторная булка N-200i"
    data-testid="ingredient_details_image_test_element"
    src="https://code.s3.yandex.net/react/code/bun-02-large.png"
  />
  <div
    className="mt-4 mb-15"
  >
    <h3
      className="text text_type_main-medium"
      data-testid="ingredient_details_name_test_element"
    >
      Краторная булка N-200i
    </h3>
    <ul
      className="content_main_info_list mt-8"
    >
      <li>
        <h4
          className="text text_type_main-default text_color_inactive mb-2"
        >
          Калории, ккал
        </h4>
        <span
          className="text text_type_digits-default text_color_inactive"
          data-testid="ingredeint_details_calories_test_element"
        >
          420
        </span>
      </li>
      <li>
        <h4
          className="text text_type_main-default text_color_inactive mb-2"
        >
          Белки, г
        </h4>
        <span
          className="text text_type_digits-default text_color_inactive"
          data-testid="ingredient_details_proteins_test_element"
        >
          80
        </span>
      </li>
      <li>
        <h4
          className="text text_type_main-default text_color_inactive mb-2"
        >
          Жиры, г
        </h4>
        <span
          className="text text_type_digits-default text_color_inactive"
          data-testid="ingredient_details_fat_test_element"
        >
          24
        </span>
      </li>
      <li>
        <h4
          className="text text_type_main-default text_color_inactive mb-2"
        >
          Углеводы, г
        </h4>
        <span
          className="text text_type_digits-default text_color_inactive"
          data-testid="ingredient_details_carbohydrates_test_element"
        >
          53
        </span>
      </li>
    </ul>
  </div>
</div>
`;
